website/content/en/docs/scorecard/custom-tests.md

```diff
@@ -0,0 +1,185 @@
+        return r
+    }
+
+    r.Log += fmt.Sprintf("Loaded ClusterServiceVersion: %s\n", csv.GetName())
+
+    ver, err := csv.GetVersion()
+    if err != nil {
+        r.Errors = append(r.Errors, err.Error())
+        r.State = scapiv1alpha2.FailState
+        return r
+    }
+
+    if ver == "" {
+        r.State = scapiv1alpha2.FailState
+        r.Errors = append(r.Errors, "csv does not have a version")
+    }
+    return r
+}
+```
+**Note**:
+The current implementation of scorecard expects the bundle under test to be mounted on the disk. The test bundle directory is present under the `testdata/bundle` folder.
+
+### Scorecard Configuration file:
+
+Modify the [config.yaml][config_yaml] file to include test definition and specify the metadata to run the test. 
+
+For the example test function `CheckVersionTest`, the following fields should be speficied in config.yaml.
+
+```yaml
+- name: "check-version"
+  image: quay.io/$USERNAME/scorecard-test:dev
+  entrypoint: 
+  - scorecard-test
+  - check-version
+  labels:
+    suite: generic
+    test: check-version-test
+  description: check if csv has version
+```
+
+The important fields to note here are:
+1. `image` - name and tag of the test image which was specified in the Makefile.
+2. `labels` - the name of the `test` and `suite` the test function belongs to. This can be specified in the `operator-sdk alpha scorecard` command to run the desired test.
+
+### Include test definition in Scorecard binary:
+
+Modify [main.go][scorecard_main_func] file under `operator-sdk/images/scorecard-test/cmd/test/` to include the test label specifier.
+
+```Go
+...     
+switch entrypoint[0] {
+case tests.OLMBundleValidationTest:
+	result = tests.BundleValidationTest(tmpDir)
+case tests.OLMCRDsHaveValidationTest:
+	result = tests.CRDsHaveValidationTest(*cfg)
+case tests.OLMCRDsHaveResourcesTest:
+	result = tests.CRDsHaveResourcesTest(*cfg)
+case tests.OLMSpecDescriptorsTest:
+	result = tests.SpecDescriptorsTest(*cfg)
+case tests.OLMStatusDescriptorsTest:
+	result = tests.StatusDescriptorsTest(*cfg)
+case tests.BasicCheckSpecTest:
+    result = tests.CheckSpecTest(*cfg)
+case tests.CheckVersionTest:
+    result = tests.CheckVersionTest(*cfg)
+default:
+	result = printValidTests()
+}
+...
+```
+
+Include custom test to the list of valid tests in the `printValidTests()` function:
+
+```Go
+func printValidTests() (result v1alpha2.ScorecardTestResult) {
+...
+    str := fmt.Sprintf("Valid tests for this image include: %s, %s, %s, %s, %s, %s",
+            tests.OLMBundleValidationTest,
+            tests.OLMCRDsHaveValidationTest,
+            tests.OLMCRDsHaveResourcesTest,
+            tests.OLMSpecDescriptorsTest,
+            tests.OLMStatusDescriptorsTest,
+            tests.BasicCheckSpecTest,
+            tests.CheckVersionTest)
+    result.Errors = append(result.Errors, str)
+...
+
+```
+
+### Run scorecard command
+
+Run the following command to execute the scorecard tests. 
+
+```console
```

Comment:
Have not explained the scorecard command here. Can include that doc as a reference in future.